test/framework/src/main/java/org/elasticsearch/test/rest/ESRestTestCase.java

```diff
@@ -91,13 +94,38 @@
     /**
      * Convert the entity from a {@link Response} into a map of maps.
      */
-    public Map<String, Object> entityAsMap(Response response) throws IOException {
+    public static Map<String, Object> entityAsMap(Response response) throws IOException {
         XContentType xContentType = XContentType.fromMediaTypeOrFormat(response.getEntity().getContentType().getValue());
-        try (XContentParser parser = createParser(xContentType.xContent(), response.getEntity().getContent())) {
+        // EMPTY and THROW are fine here because `.map` doesn't use named x content or deprecation
+        try (XContentParser parser = xContentType.xContent().createParser(
+                NamedXContentRegistry.EMPTY, DeprecationHandler.THROW_UNSUPPORTED_OPERATION,
+                response.getEntity().getContent())) {
             return parser.map();
         }
     }
 
+    /**
+     * Does the cluster being tested have xpack installed?
+     */
+    public static boolean hasXPack() throws IOException {
+        RestClient client = adminClient();
+        if (client == null) {
+            throw new IllegalStateException("must be called inside of a rest test case test");
+        }
+        Map<?, ?> response = entityAsMap(client.performRequest(new Request("GET", "_nodes/plugins")));
+        Map<?, ?> nodes = (Map<?, ?>) response.get("nodes");
```

Comment:
Do we need to do any null checks here? Im a-ok not doing it given we always know it will come back w/ the data we want/need. I know its just a call to `_nodes/plugins` so I assume this is legit.